addons/pos_gift_card/static/src/js/models.js

```diff
@@ -4,65 +4,61 @@ odoo.define("pos_gift_card.gift_card", function (require) {
-        return model.model === 'product.product';
-    });
-    var product_model = existing_models[product_index];
+  Registries.PosModelRegistry.extend(models.PosGlobalState, (PosGlobalState) => {
 
-  models.load_models([
-    {
-      model: "gift.card",
-      loaded: function (self, giftCard) {
-        self.giftCard = giftCard;
-      },
-    },
-  ]);
+    class PosGiftCardPosGlobalState extends PosGlobalState {
+      async _processData(loadedData) {
+        await super._processData(...arguments);
+        this.giftCards = loadedData['gift.card'];
+      }
+    }
+
+    return PosGiftCardPosGlobalState;
+  });
 
   Registries.PosModelRegistry.extend(models.Order, (Order) => {
 
-  class PosGiftCardOrder extends Order {
-    set_orderline_options(orderline, options) {
-      super.set_orderline_options(...arguments);
-      if (options && options.generated_gift_card_ids) {
-        orderline.generated_gift_card_ids = [options.generated_gift_card_ids];
+    class PosGiftCardOrder extends Order {
+      set_orderline_options(orderline, options) {
+        super.set_orderline_options(...arguments);
+        if (options && options.generated_gift_card_ids) {
+          orderline.generated_gift_card_ids = [options.generated_gift_card_ids];
+        }
+        if (options && options.gift_card_id) {
+          orderline.gift_card_id = options.gift_card_id;
+        }
       }
-      if (options && options.gift_card_id) {
-        orderline.gift_card_id = options.gift_card_id;
+      wait_for_push_order() {
+          if(this.pos.config.use_gift_card) {
+              let giftProduct = this.pos.db.product_by_id[this.pos.config.gift_card_product_id[0]];
+              for (let line of this.orderlines) {
+                  if(line.product.id === giftProduct.id)
+                      return true;
+              }
+          }
+          return super.wait_for_push_order(...arguments);
       }
     }
-    wait_for_push_order() {
-        if(this.pos.config.use_gift_card) {
-            let giftProduct = this.pos.db.product_by_id[this.pos.config.gift_card_product_id[0]];
-            for (let line of this.orderlines) {
-                if(line.product.id === giftProduct.id)
-                    return true;
-            }
-        }
-        return super.wait_for_push_order(...arguments);
-    }
-  }
 
-  return PosGiftCardOrder;
+    return PosGiftCardOrder;
   });
 
   Registries.PosModelRegistry.extend(models.Orderline, (Orderline) => {
 
-  class PosGiftCardOrderline extends Orderline {
-    export_as_JSON() {
-      var json = super.export_as_JSON(...arguments);
-      json.generated_gift_card_ids = this.generated_gift_card_ids;
-      json.gift_card_id = this.gift_card_id;
-      return json;
-    }
-    init_from_JSON(json) {
-      super.init_from_JSON(...arguments);
-      this.generated_gift_card_ids = json.generated_gift_card_ids;
-      this.gift_card_id = json.gift_card_id;
+    class PosGiftCardOrderline extends Orderline {
```

Comment:
same here: indentation of class is not the same as in other module, de-indent it.